packages/wallet/ui/src/util/fixed-websocket-connector.js

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable no-underscore-dangle */
+import { E } from '@endo/eventual-send';
+
+const CONNECTION_TIMEOUT_MS = 5000;
+
+// Wait for the wallet to finish loading.
+export const waitForBootstrap = async getBootstrap => {
+  const getLoadingUpdate = (...args) =>
+    E(E.get(getBootstrap()).loadingNotifier).getUpdateSince(...args);
+  let update = await getLoadingUpdate();
+  while (update.value.includes('wallet')) {
+    console.log('waiting for wallet');
+    // eslint-disable-next-line no-await-in-loop
+    update = await getLoadingUpdate(update.updateCount);
+  }
+
+  return getBootstrap();
+};
+
+export const makeFixedWebSocketConnector = href => component => {
+  let ws;
+
+  const onAdminOpen = () => {
+    const send = obj => {
+      if (ws.readyState !== WebSocket.OPEN) {
+        return;
+      }
+      ws.send(JSON.stringify(obj));
+    };
+    component._startCapTP(send, 'walletAdmin');
+
+    assert(component._captp);
+    const { dispatch, getBootstrap } = component._captp;
+
+    ws.addEventListener('message', ev => {
+      const obj = JSON.parse(ev.data);
+      dispatch(obj);
+    });
+
+    const adminFacet = E(
+      E.get(waitForBootstrap(getBootstrap)).wallet,
+    ).getAdminFacet();
+    adminFacet.then(component.service.send({ type: 'connected' }));
+    component._bridgePK.resolve(adminFacet);
+  };
+
+  return {
+    render: () => '',
```

Comment:
no need to be a string
```suggestion
    render: () => null,
```

But why is this a React component at all if it never renders?